File: src/test/java/TestAddition.java
import org.junit.Assert;
import org.junit.BeforeClass;
import org.junit.Test;

public class TestAddition {

    private static Fraction fraction;
    public static int numerator = 15;
    public static int denominator = 5;

    @BeforeClass
    public static void createNewFraction() {
        fraction = new Fraction(numerator, denominator);
    }


    @Test(expected = ArithmeticException.class)
    public void zeroDenominatorShouldThrowException() {
        fraction.dividedBy(new Fraction(5, 0));
    }

    @Test
    public void testSimple() {

        var result = fraction.plus(new Fraction(3, 5));

        Assert.assertEquals((Integer) 18, result.getNumerator());
        Assert.assertEquals((Integer) 5, result.getDenominator());
    }


    @Test
    public void testNormalisation() {
        var nFraction = new Fraction(1,2);
        var result = nFraction.plus(new Fraction(1, 2));

        Assert.assertEquals((Integer) 1, result.getNumerator());
        Assert.assertEquals((Integer) 1, result.getDenominator());
    }

    @Test
    public void divideTest() {
        var result = fraction.dividedBy(new Fraction(3, 5));

        Assert.assertEquals((Integer) 5, result.getNumerator());
        Assert.assertEquals((Integer) 1, result.getDenominator());
    }

    @Test
    public void timesTest() {
        var result = fraction.times(new Fraction(3, 5));

        Assert.assertEquals((Integer) 9, result.getNumerator());
        Assert.assertEquals((Integer) 5, result.getDenominator());
    }

    @Test
    public void minusTest() {
        var result = fraction.minus(new Fraction(3, 5));

        Assert.assertEquals((Integer) 12, result.getNumerator());
        Assert.assertEquals((Integer) 5, result.getDenominator());
    }
}

File: src/test/java/TestAddition.kt
import org.junit.Assert
import java.lang.ArithmeticException
import org.junit.BeforeClass
import org.junit.Test

class TestAddition {
    @Test(expected = ArithmeticException::class)
    fun zeroDenominatorShouldThrowException() {
        fraction!!.dividedBy(Fraction(5, 0))
    }

    @Test
    fun testSimple() {
        val result = fraction!!.plus(Fraction(3, 5))
        Assert.assertEquals(18, result.numerator)
        Assert.assertEquals(5, result.denominator)
    }

    @Test
    fun testNormalisation() {
        val nFraction = Fraction(1, 2)
        val result = nFraction.plus(Fraction(1, 2))
        Assert.assertEquals(1, result.numerator)
        Assert.assertEquals(1, result.denominator)
    }

    @Test
    fun divideTest() {
        val result = fraction!!.dividedBy(Fraction(3, 5))
        Assert.assertEquals(5, result.numerator)
        Assert.assertEquals(1, result.denominator)
    }

    @Test
    fun timesTest() {
        val result = fraction!!.times(Fraction(3, 5))
        Assert.assertEquals(9, result.numerator)
        Assert.assertEquals(5, result.denominator)
    }

    @Test
    fun minusTest() {
        val result = fraction!!.minus(Fraction(3, 5))
        Assert.assertEquals(12, result.numerator)
        Assert.assertEquals(5, result.denominator)
    }

    companion object {
        private var fraction: Fraction? = null
        var numerator = 15
        var denominator = 5
        @BeforeClass
        fun createNewFraction() {
            fraction = Fraction(numerator, denominator)
        }
    }
}